diff --git a/kded/ui/vaultcreationwizard.cpp b/kded/ui/vaultcreationwizard.cpp
index ff6aae7..66b0fd8 100644
--- a/kded/ui/vaultcreationwizard.cpp
+++ b/kded/ui/vaultcreationwizard.cpp
@@ -83,31 +83,5 @@ public:
                 step { directoryPairChooser(DirectoryPairChooserWidget::RequireEmptyDirectories) },
                 step { activitiesChooser() }
             }
-        },
-
-        { "cryfs" / i18n("CryFS"),
-            {
-                step { notice("cryfs-message",
-                       i18n("<b>Security notice:</b>\n\
-                             CryFS encrypts your files, so you can safely store them anywhere.\n\
-                             It works well together with cloud services like Dropbox, iCloud, OneDrive and others.\n\
-                             <br /><br />\n\
-                             Unlike some other file-system overlay solutions,\n\
-                             it does not expose the directory structure,\n\
-                             the number of files nor the file sizes\n\
-                             through the encrypted data format.\n\
-                             <br /><br />\n\
-                             One important thing to note is that,\n\
-                             while CryFS is considered safe,\n\
-                             there is no independent security audit\n\
-                             which confirms this."))
-                     },
-                step { passwordChooser() },
-                step { directoryPairChooser(DirectoryPairChooserWidget::RequireEmptyDirectories) },
-                step {
-                    cryfsCypherChooser(),
-                    activitiesChooser()
-                }
-            }
         }
     };
