diff --git a/ges/meson.build b/ges/meson.build
index fe6ca740..bd5c14e8 100644
--- a/ges/meson.build
+++ b/ges/meson.build
@@ -136,7 +136,7 @@ version_data.set('GES_VERSION_NANO', gst_version_nano)
 
 configure_file(input : 'ges-version.h.in',
   output : 'ges-version.h',
-  install_dir : 'include/gstreamer-1.0/ges',
+  install_dir : get_option('includedir') + '/gstreamer-1.0/ges',
   configuration : version_data)
 
 install_headers(ges_headers, subdir : 'gstreamer-1.0/ges')
